keras/backend/tensorflow/numpy.py

```diff
@@ -71,7 +71,21 @@ def bincount(x, weights=None, minlength=0):
 
 
 def einsum(subscripts, *operands, **kwargs):
-    return tfnp.einsum(subscripts, *operands, **kwargs)
+    operands = tf.nest.map_structure(convert_to_tensor, operands)
+
+    dtypes_to_resolve = []
+    for x in operands:
+        dtypes_to_resolve.append(x.dtype)
+    result_dtype = dtypes.result_type(*dtypes_to_resolve)
+    compute_dtype = result_dtype
+    # TODO: tfnp.einsum doesn't support integer dtype with gpu
+    if "int" in compute_dtype:
+        compute_dtype = config.floatx()
+
+    operands = tf.nest.map_structure(
+        lambda x: tf.cast(x, compute_dtype), operands
+    )
+    return tf.cast(tfnp.einsum(subscripts, *operands, **kwargs), result_dtype)
```

Comment:
Can we just use `tf.einsum` directly? Any API difference?

Reply:
I have replaced tfnp.einsum with tf.einsum. However, the dtype-related logic can't be simplified since tfnp.einsum is essentially a wrapped version of tf.einsum.